hf_xet/src/lib.rs

```diff
@@ -106,53 +140,129 @@ fn try_parse_progress_updaters(funcs: Vec<Py<PyAny>>) -> PyResult<Vec<Arc<dyn Pr
     Ok(updaters)
 }
 
-#[pyclass]
+// TODO: we won't need to subclass this in the next major version update.
+#[pyclass(subclass)]
 #[derive(Clone, Debug)]
-pub struct PyPointerFile {
+pub struct PyXetDownloadInfo {
     #[pyo3(get, set)]
-    path: String,
+    destination_path: String,
     #[pyo3(get)]
     hash: String,
     #[pyo3(get)]
-    filesize: u64,
+    file_size: u64,
 }
 
-impl From<PointerFile> for PyPointerFile {
-    fn from(pf: PointerFile) -> Self {
+#[pymethods]
+impl PyXetDownloadInfo {
+    #[new]
+    pub fn new(destination_path: String, hash: String, file_size: u64) -> Self {
         Self {
-            path: pf.path().to_string(),
-            hash: pf.hash_string().to_string(),
-            filesize: pf.filesize(),
+            destination_path,
+            hash,
+            file_size,
         }
     }
-}
 
-impl From<PyPointerFile> for PointerFile {
-    fn from(pf: PyPointerFile) -> Self {
-        PointerFile::init_from_info(&pf.path, &pf.hash, pf.filesize)
+    fn __str__(&self) -> String {
+        format!("{self:?}")
+    }
+
+    fn __repr__(&self) -> String {
+        format!("PyXetDownloadInfo({}, {}, {})", self.destination_path, self.hash, self.file_size)
     }
 }
 
+// TODO: on the next major version update, delete this class and the trait implementation.
+// This is used to support backward compatibility for PyPointerFile with old versions of huggingface_hub
+#[pyclass(extends=PyXetDownloadInfo)]
+#[derive(Clone, Debug)]
+pub struct PyPointerFile {}
+
 #[pymethods]
 impl PyPointerFile {
     #[new]
-    pub fn new(path: String, hash: String, filesize: u64) -> Self {
-        Self { path, hash, filesize }
+    pub fn new(path: String, hash: String, filesize: u64) -> (Self, PyXetDownloadInfo) {
+        (PyPointerFile {}, PyXetDownloadInfo::new(path, hash, filesize))
+    }
+
+    fn __str__(&self) -> String {
+        format!("{self:?}")
+    }
+
+    fn __repr__(self_: PyRef<'_, Self>) -> String {
+        let super_ = self_.as_super();
+        format!("PyPointerFile({}, {}, {})", super_.destination_path, super_.hash, super_.file_size)
+    }
+
+    fn get_path(self_: PyRef<'_, Self>) -> String {
+        self_.as_super().destination_path.clone()
+    }
+
+    fn get_hash(self_: PyRef<'_, Self>) -> String {
```

Comment:
This isn't needed since hash is unchanged from the parent class.

However, you do need a `getter` for `filesize` to keep this backwards-compatible.